integration/querydsl/blaze-persistence-querydsl-testsuite/src/test/java/com/blazebit/persistence/querydsl/BasicQueryTest.java

```diff
@@ -0,0 +1,744 @@
+                            .limit(1), subT2)
+                    .on(t.eq(subT2))
+                    .fetch();
+
+            assertNotNull(fetch);
+
+        });
+    }
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOracle.class, NoOpenJPA.class, NoHibernate42.class, NoHibernate43.class, NoHibernate50.class, NoHibernate51.class, NoH2.class, NoMySQLOld.class })
+    public void testMultipleInlineEntityWithLimitJoinLateral() {
+        Assume.assumeFalse(dbmsDialect.getLateralStyle() == LateralStyle.NONE);
+
+        doInJPA(entityManager -> {
+            QRecursiveEntity t = new QRecursiveEntity("t");
+            QRecursiveEntity subT = new QRecursiveEntity("subT");
+            QRecursiveEntity subT2 = new QRecursiveEntity("subT2");
+            QRecursiveEntity subT3 = new QRecursiveEntity("subT3");
+
+            new BlazeJPAQuery<RecursiveEntity>(entityManager, cbf)
+                    .select(t)
+                    .from(new BlazeJPAQuery<RecursiveEntity>().select(t).from(t)
+                            .leftJoin(selectFrom(subT).where(subT.parent.name.eq("root1")).orderBy(subT.name.asc()).limit(1), subT)
+                            .on(t.eq(subT))
+                            .where(t.parent.name.eq("root1"))
+                            .orderBy(t.name.asc())
+                            .limit(1L), t)
+                    .leftJoin(selectFrom(subT2)
+                        .where(subT2.parent.name.eq("root1"))
+                        .orderBy(subT2.name.asc())
+                        .limit(1), subT3)
+                        .on(t.eq(subT3))
+                        .lateral()
+                    .fetch();
+
+        });
+    }
+
+    // NOTE: Entity joins are only supported on Hibernate 5.1+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoHibernate42.class, NoHibernate43.class, NoHibernate50.class, NoH2.class, NoMySQLOld.class, NoOracle.class })
+    public void testJoinInlineEntityWithLimit() {
+        Assume.assumeFalse(dbmsDialect.getLateralStyle() == LateralStyle.NONE);
+
+        doInJPA(entityManager -> {
+            QRecursiveEntity t = new QRecursiveEntity("t");
+            QRecursiveEntity subT = new QRecursiveEntity("subT");
+            QRecursiveEntity subT2 = new QRecursiveEntity("subT2");
+
+            List<Tuple> fetch = new BlazeJPAQuery<RecursiveEntity>(entityManager, cbf)
+                    .select(t, subT2)
+                    .from(t)
+                    .leftJoin(select(subT).from(t.children, subT).orderBy(subT.id.asc()).limit(1), subT2)
+                    .lateral()
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoHibernate42.class, NoHibernate43.class, NoHibernate50.class, NoHibernate51.class, NoH2.class, NoMySQLOld.class, NoOracle.class })
```

Comment:
Please add comments for the exclusions. The Oracle and old MySQL version we test with doesn't support lateral, H2 also doesn't.